src/link/edit.js

```diff
@@ -102,6 +108,50 @@ const Edit = ( {
 		}
 	};
 
+	const getFormatStart = ( position ) => {
+		if ( value.formats[ position ] && value.formats[ position ][ 0 ].type === formatType ) {
+			return getFormatStart( position - 1 );
+		}
+		return position;
+	};
+
+	const getFormatEnd = ( position ) => {
+		if ( value.formats[ position ] && value.formats[ position ][ 0 ].type === formatType ) {
+			return getFormatEnd( position + 1 );
+		}
+		return position;
+	};
+
+	const getTrimmedStart = ( selectedValue ) => {
+		const selectedString = selectedValue.text.slice( selectedValue.start, selectedValue.end );
+		const trimmed = selectedString.trimStart();
+		if ( selectedString.length !== trimmed.length ) {
+			const delta = selectedString.length - trimmed.length;
+			return selectedValue.start + delta;
+		}
+		return selectedValue.start;
+	};
+
+	const getTrimmedEnd = ( selectedValue ) => {
+		const selectedString = selectedValue.text.slice( selectedValue.start, selectedValue.end );
+		const trimmed = selectedString.trimEnd();
+		if ( selectedString.length !== trimmed.length ) {
+			const delta = selectedString.length - trimmed.length;
+			return selectedValue.end - delta;
+		}
+		return selectedValue.end;
+	};
+
+	const handleTextEdit = () => {
+		const editDetected = lastValue.formats.length !== value.formats.length;
```

Comment:
i think it should check the entire string 

for example i change `cat` to `cot`, it doesn't detect an edit

Reply:
Good call, done